tests/onnxruntime/test_modeling.py

```diff
@@ -251,6 +254,58 @@ def test_model_on_cpu_str(self):
         self.assertEqual(model.device, cpu)
         self.assertListEqual(model.providers, ["CPUExecutionProvider"])
 
+    def test_missing_execution_provider(self):
+        with self.assertRaises(ValueError) as cm:
+            model = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider="ThisProviderDoesNotExist")
+
+        self.assertTrue("but the available execution providers" in str(cm.exception))
+
+        is_onnxruntime_gpu_installed = (
+            subprocess.run("pip list | grep onnxruntime-gpu", shell=True, capture_output=True).stdout.decode("utf-8")
+            != ""
+        )
+        is_onnxruntime_installed = "onnxruntime " in subprocess.run(
+            "pip list | grep onnxruntime", shell=True, capture_output=True
+        ).stdout.decode("utf-8")
+
+        if not is_onnxruntime_gpu_installed:
+            for provider in ["CUDAExecutionProvider", "TensorrtExecutionProvider"]:
+                with self.assertRaises(ImportError) as cm:
+                    _ = ORTModel.from_pretrained(self.ONNX_MODEL_ID, provider=provider)
+
+                self.assertTrue(
+                    f"Asked to use {provider}, but `onnxruntime-gpu` package was not found." in str(cm.exception)
+                )
+        else:
+            logger.info("Skipping CUDAExecutionProvider/TensorrtExecutionProvider without `onnxruntime-gpu` test")
+
+        # need to install first onnxruntime-gpu, then onnxruntime for this test to pass,
+        # thus overwritting onnxruntime/capi/_ld_preload.py
+        if is_onnxruntime_installed and is_onnxruntime_gpu_installed:
```

Comment:
Are you testing that double installation of onnxruntime and onnxruntime-gpu won't work for Cuda ep and TRT ep? Why should we add this test?

Reply:
We check that in case the onnxruntime installation is messed up, a meaningful error message is raised.